### PR TITLE
[Optimization] support mm prefill batch

### PR DESCRIPTION
**Pull request by kevincheng2**
## Motivation

> :bulb: If this PR is a Cherry Pick, the PR title needs to follow the format by adding the [Cherry-Pick] label at the very beginning and appending the original PR ID at the end. For example, [Cherry-Pick][CI] Add check trigger and logic(#5191)

> :bulb: 如若此PR是Cherry Pick，PR标题需遵循格式，在最开始加上[Cherry-Pick]标签，以及最后面加上原PR ID，例如[Cherry-Pick][CI] Add check trigger and logic(#5191) 

- 支持多模 prefill 多batch，改动影响模型包括ernie_vl系列、qwen_vl系列、paddleocr_vl、pooling_model(vl)
- paddleocr_vl 模型支持prefix caching、encoder cache
- 修复encoder cache 未生效的bug
- 保留eb5支持多模prefill单batch+纯文prefill多batch逻辑

## Modifications

1. 移除逐请求处理rope_3d_position_ids和 image_features的 `get_chunked_inputs`  方法，将多模 rope_3d_position_ids和 image_features的计算统一迁移到 `_process_mm_features ` 方法中，以batch为单位统一处理
2. 增加 `_get_feature_positions ` 方法支持对 image_feature 切chunk，同时支持扩展图片均匀分chunk场景
3. 移除调度侧和paddle_ocr对多模prefill batch的限制

## Usage or Command

无需配置，默认生效

## Checklist

- [x] Add at least a tag in the PR title.
  - Tag list: [`[FDConfig]`,`[APIServer]`,`[Engine]`, `[Scheduler]`, `[PD Disaggregation]`, `[Executor]`, `[Graph Optimization]`, `[Speculative Decoding]`, `[RL]`, `[Models]`, `[Quantization]`, `[Loader]`, `[OP]`, `[KVCache]`, `[DataProcessor]`, `[BugFix]`, `[Docs]`, `[CI]`, `[Optimization]`, `[Feature]`, `[Benchmark]`, `[Others]`, `[XPU]`, `[HPU]`, `[GCU]`, `[DCU]`, `[Iluvatar]`, `[Metax]`]
  - You can add new tags based on the PR content, but the semantics must be clear.
- [x] Format your code, run `pre-commit` before commit.
- [x] Add unit tests. Please write the reason in this PR if no unit tests.
- [x] Provide accuracy results.
- [x] If the current PR is submitting to the `release` branch, make sure the PR has been submitted to the `develop` branch, then cherry-pick it to the `release` branch with the `[Cherry-Pick]` PR tag.

### CODE REVIEW

File: fastdeploy/worker/xpu_model_runner.py
Comment (ddchenhao66) [inline fastdeploy/worker/xpu_model_runner.py:1]: xpu_model_runner里batch_uncached_inputs、scatter_and_cache_features、extract_vision_features等和gpu_model_runner里逻辑一致的函数不需要修改吗？

Comment (kevincheng2) [reply fastdeploy/worker/xpu_model_runner.py:1]: 不需要，现在xpu还没有支持prefill 多batch，所以还是沿用之前的逻辑，调度层没有修改，不会影响xpu

File: fastdeploy/engine/sched/resource_manager_v1.py [hunk 1]
```diff
@@ -655,7 +655,7 @@ def _allocate_decode_and_extend():
 
                     request = self.waiting[0]
                     if (
-                        not envs.FD_ENABLE_MAX_PREFILL
+                        self.config.model_config.disable_mm_prefill_batch()
```
Comment (TBD1) [inline fastdeploy/engine/sched/resource_manager_v1.py:658]: 这里是兼容eb5吗，非eb5默认开启多batch？

Comment (kevincheng2) [reply fastdeploy/engine/sched/resource_manager_v1.py:658]: 是的，现在只有eb5 还是保持之前的组batch逻辑：纯文多batch + 多模单batch，这里后边可以一起看下，尽量都统一成多batch

File: fastdeploy/worker/gpu_model_runner.py [hunk 2]
```diff
@@ -375,188 +376,242 @@ def _init_logits_processor(self, request) -> tuple[Future[LogitsProcessorBase],]
...
+                    if envs.FD_ENABLE_MAX_PREFILL:
+                        if "vit_seqlen" in inputs:
+                            vit_seqlen_list = inputs["vit_seqlen"][request.num_image_start : request.num_image_end]
+                        if "vit_position_ids" in inputs:
+                            vit_position_ids_list = inputs["vit_position_ids"][
+                                request.num_image_start : request.num_image_end
+                            ]
+                    grid_thw_list = inputs["grid_thw"][request.num_image_start : request.num_image_end]
+                    mm_hashes_list = inputs["mm_hashes"][request.num_image_start : request.num_image_end]
+                    feature_positions = self._get_feature_positions(
```
Comment (TBD1) [inline fastdeploy/worker/gpu_model_runner.py:445]: 这里mm_positions为啥要再切chunk，之前的处理逻辑是不是都在resource_manager_v1里面和token一起切分

Comment (kevincheng2) [reply fastdeploy/worker/gpu_model_runner.py:445]: 这里表示是每个feature在当前推理的这段请求中，要不要被切开(像cache命中部分featue或多模均匀分chunk的场景用到的)

**Review by TBD1 — APPROVED** [review]
